src/AudioSystem.cpp

```diff
@@ -28,17 +41,29 @@ AudioSystem::AudioSystem() {
 
 void AudioSystem::load_audio_assets() {
     // Load all sound assets
-    sound_assets_["beat"] = Mix_LoadWAV("assets/audio/beat.wav");
-
-    if (sound_assets_.at("beat") == nullptr) {
-        std::cerr << "Failed to load sound: beat.wav" << std::endl;
+    for (auto& sound_asset_info : SOUND_ASSETS_INFO) {
+        const char* key = sound_asset_info.first;
+        const char* path = sound_asset_info.second;
+        sound_assets_[key] = Mix_LoadWAV(path);
+
+        if (sound_assets_.at(key) == nullptr) {
+            std::cerr << "Failed to load sound: " << key << std::endl;
+            init_successful_ = false;
```

Comment:
Im actually not a fan of this `init_successful_` pattern. I think it would be better and easier to use if functions explicitly return false and are dealt with there, instead of a state being managed.

Reply:
Yeah that's fair. I'll go change it to that